pkg/segment/writer/agiletree.go

```diff
@@ -41,6 +48,48 @@ const (
 	TotalMeasFns
 )
 
+var STBHolderPool [MaxConcurrentAgileTrees]*STBHolder
+
+type STBHolder struct {
+	stbPtr            *StarTreeBuilder
+	currentlyInUse    bool
+	lastUsedTimestamp time.Time
+}
+
+func GetSTB() *STBHolder {
+	atreeCounterLock.Lock()
+	defer atreeCounterLock.Unlock()
+	if currentAgileTreeCount == MaxConcurrentAgileTrees {
```

Comment:
nit: I think what you have here is fine. But being too strict on the equality check is bit nervous. Just in case if there are bugs in rest of the code base and the`currentAgileTreeCount` goes past `MaxConcurrentAgileTrees` then we will keep adding trees. In such scenarios I am usually follow 
`currentAgileTreeCount >= MaxConcurrentAgileTrees` pattern to be doubly sure about this.
But again up to you if you want to keep it this way or switch to `>=`

Reply:
addressed in the latest commit.